Code/Data_science_project_exploration.Rmd

```diff
@@ -0,0 +1,280 @@
+final_forecast_period <- forecastdemand_nsw %>% 
+  group_by(DATETIME) %>% 
+  summarise_at(vars(PERIODID),
+               list(PERIODID = max))
+final_forecast <- forecastdemand_nsw %>%
+  inner_join(final_forecast_period,by=c('DATETIME','PERIODID')) %>%
+  select(DATETIME,
+         FORECASTDEMAND) %>%
+  rename('FINAL_FORECAST'='FORECASTDEMAND')
+
+##join initial and final forecast to table data
+data <- data %>% 
+  left_join(initial_forecast,by=c('DATETIME')) %>%
+  left_join(final_forecast,by=c('DATETIME'))
+
+##fill in missing initial/final forecast demand (down fill)
+data <- data %>% 
+  fill(INITIAL_FORECAST) %>%
+  fill(FINAL_FORECAST)
+
+##Create DATE and MONTHDATE field for plotting
+data$DATE <- as.Date(substr(data$DATETIME,1,10))
+data$YEARMONTH<- as.Date(paste(substr(data$DATE,1,7), "-01", sep=""))
+  
+##convert DATETIME to date
+data$DATETIME <- ymd_hms(data$DATETIME)
+
+##remove unneeded table
+rm('final_forecast_period')
+
+```
+
+Generate relevant date time derivations.
+
+```{r create date fields}
+##create year/month/day/hour/minute fields
+data <- data %>% mutate(YEAR=year(DATETIME), 
+                        MONTH=month(DATETIME),
+                        DAY=day(DATETIME),
+                        HOUR=hour(DATETIME), 
+                        MINUTE=minute(DATETIME))
+
+##season
+data <- data %>%mutate(SEASON = case_when(
+  MONTH %in%  9:11 ~ "SPRING",
+  MONTH %in%  c(12, 1, 2)  ~ "SUMMER",
+  MONTH %in%  3:5  ~ "AUTUMN",
+  TRUE ~ "WINTER"))
+##create boolean for 8am~8pm timeslot
+data <- data %>%mutate(timeframe_8amto8pm = case_when(
+  HOUR %in%  8:20 ~ TRUE,
+  TRUE ~ FALSE))
+##Get days of week, monday = 1,...,sunday=7
+data<- data%>%mutate(DaysOfWeek=wday(DATETIME,
+                                     week_start = getOption("lubridate.week.start", 1)))
+##get weekday&weekend
+data <- data %>%mutate(WEEKDAYWEEKEND = case_when(
+  DaysOfWeek %in%  1:5 ~ "WEEKDAY",
+  DaysOfWeek %in%  6:7  ~ "WEEKEND",
+))
+
+##Deduplicate
+data <- data %>% distinct()
+
+```
+
+Calculate rolling datetime difference to get gaps, and summarise
+```{r Get rolling time difference}
+data <- data %>%  mutate(DATETIME_DIFF = DATETIME - lag(DATETIME))
+table(data$DATETIME_DIFF)
+
+```
+investigate the 0 min diff rows (resolved):
+```{r}
+##no more duplicates
+data %>% filter(DATETIME_DIFF==0)
+```
+
+Plot data for datetime against total demand:
+```{r plot }
+plot(data$TOTALDEMAND~data$DATETIME,type='l')
+
+```
+
+Due to the number of spikes, it is difficult to identify trends, attempting to plot by year-month, aggregating energy usage by averaging (since time interval is not consistent):
+```{r plot2}
+##drop incomplete period
+##Calculate grouped total demand and forecast
+data_YM_plot <- data %>% 
+  filter(YEARMONTH != as.Date('2022-08-01')) %>%
+  group_by(YEARMONTH) %>%
+  summarise(T_DEMAND=mean(TOTALDEMAND),
+            T_forecastfinal = mean(FINAL_FORECAST))
+
+##First row of March (Autumn)
+data_YM_plot <- data_YM_plot %>%
```

Comment:
[no change required]; these four blocks of code could be funcionalised

Reply:
Agreed, I'll do the finetuning